packages/grafana-ui/src/components/Layout/Box/Box.tsx

```diff
@@ -68,65 +68,74 @@ interface BoxProps {
   element?: ElementType;
 }
 
-export const Box = ({
-  children,
-  margin,
-  marginX,
-  marginY,
-  marginTop,
-  marginBottom,
-  marginLeft,
-  marginRight,
-  padding,
-  paddingX,
-  paddingY,
-  paddingTop,
-  paddingBottom,
-  paddingLeft,
-  paddingRight,
-  display,
-  backgroundColor,
-  grow,
-  shrink,
-  borderColor,
-  borderStyle,
-  borderRadius,
-  justifyContent,
-  alignItems,
-  boxShadow,
-  element,
-}: React.PropsWithChildren<BoxProps>) => {
-  const styles = useStyles2(
-    getStyles,
-    margin,
-    marginX,
-    marginY,
-    marginTop,
-    marginBottom,
-    marginLeft,
-    marginRight,
-    padding,
-    paddingX,
-    paddingY,
-    paddingTop,
-    paddingBottom,
-    paddingLeft,
-    paddingRight,
-    display,
-    backgroundColor,
-    grow,
-    shrink,
-    borderColor,
-    borderStyle,
-    borderRadius,
-    justifyContent,
-    alignItems,
-    boxShadow
-  );
-  const Element = element ?? 'div';
+export const Box = forwardRef<HTMLElement, PropsWithChildren<BoxProps>>(
+  (
+    {
```

Comment:
It's not an established rule (yet 😈 ), but do you think we could not destructure the props in the function argument?

instead do like

```
forwardRef<...>(props, ref) {
  const { children, margin, etc } = props;
```

just to improve readability a bit 👼🏻

Reply:
Updated the code, don't think it has any impact on the readability though, especially with this number of props.

Reply:
Yeah, i get what you're saying.

The main struggle i have is specifically with reading the arguments to the forwardRef callback. I think this makes _that_ easier (though, yes, the eventual unpacking is stuff rough)